utils/s2n_safety.c

```diff
@@ -171,3 +171,18 @@ int s2n_mul_overflow(uint32_t a, uint32_t b, uint32_t* out)
     *out = (uint32_t) result;
     return S2N_SUCCESS;
 }
+
+int s2n_align_to(uint32_t initial, uint32_t alignment, uint32_t* out)
```

Comment:
There's other places where we do alignment, like aligning to cryptographic block sizes, but this function wouldn't be safe in some contexts because it's not constant time. Should we name the function defensively? or am I being too paranoid.

Reply:
Is there a constant time alignment you do there?
If you know that the alignment is a power of 2, we can do it with bit tweaks - would that be a safe restriction?

Reply:
The padding for CBC is the main place, but it's all done in one pass and it's a pretty special case; I wouldn't want to use this function there, but that's ok. The name seems fine the more I think about it.

Reply:
I've played around with bit-twiddling.  Does this make sense? It only works if alignment is a power of 2.

```
    S2N_PRECONDITION(alignment != 0);
    uint32_t mask = alignment - 1;
    uint32_t upper = initial & ~mask;
    uint32_t lower = initial & mask;
    uint32_t add_mask = 0 - (lower == 0);
    uint64_t result = upper + (alignment & add_mask);
    S2N_ERROR_IF(result > UINT32_MAX, S2N_ERR_INTEGER_OVERFLOW);
```

Reply:
https://github.com/awslabs/s2n/issues/1586 to track this

Reply:
> Should we name the function defensively?

I don't know if it's just me, but I assume functions are NOT constant time unless they have `constant_time` in their name like `s2n_constant_time_equals()`, `s2n_constant_time_copy_or_dont()`, and `s2n_constant_time_pkcs1_unpad_or_dont()`.